### PR TITLE
Add initialization logic for multi-worker/process distribution.

### PR DESCRIPTION
**Pull request by qlzh727**
This should address part 1 in #18560

### CODE REVIEW

**Review by fchollet — COMMENTED** [review]
Thank you for the PR!

File: keras/distribution/distribution_lib.py [hunk 1]
```diff
@@ -40,6 +41,88 @@ def list_devices(device_type=None):
...
+    For certain backend that also rely on the enviornment variables to
+    configure, Keras will properly forward them.
+
+    Args:
+      job_addresses: string. Comma separated IP addresses for all the jobs that
+        will form the whole computation cluster. Note that for JAX backend, only
+        the address for job 0 (coodinator) is needed. For certain runtime like
+        cloud TPU, this value can be `None`, and the backend will figure it out
+        with the TPU enviornment variables. You can also config this value via
+        enviornment variable `KERAS_DISTRIBUTE_JOB_ADDRESSES`.
```
[the remark this answers is not in the document]
Comment (qlzh727) [reply keras/distribution/distribution_lib.py]: Done.

File: keras/distribution/distribution_lib.py [hunk 2]
```diff
@@ -40,6 +41,88 @@ def list_devices(device_type=None):
...
+        configure this value via enviornment variable
+        `KERAS_DISTRIBUTE_PROCESS_ID`.
+
+    Example:
+      Suppose there are two GPU processes, and process 0 is running at
+      address `10.0.0.1:1234`, and process 1 is running at address
+      `10.0.0.2:2345`. To configure such cluster, you can run
+
+      On process 0:
+      ```
```
[the remark this answers is not in the document]
Comment (qlzh727) [reply keras/distribution/distribution_lib.py]: Done.

**Review by fchollet — APPROVED** [review]
LGTM, thank you!

**Pull request merged**